From 9f17e80eebb999b3ab89e079058d3bec0457519c Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 28 Apr 2015 14:41:36 +0800
Subject: [PATCH 0110/5242] MLK-10718 usb: chipidea: udc: disable BSV irq when
 otg from undefined to a_idle

commit  e59f5681b0a90c881216e27b8e190e4759daa962 from
https://source.codeaurora.org/external/imx/linux-imx.git

After device power up, insert usb OTG ID cable, otg fsm from undefined state
to a_idle, the BSV cannot be disabled, this will result BSV irq generated but
which is actually only for B device.

Signed-off-by: Li Jun <jun.li@freescale.com>
(cherry picked from commit bf6a1cdaef68125cc10240f174019e2646ebaa24)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/udc.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/chipidea/udc.c b/drivers/usb/chipidea/udc.c
index bcfed3f..63b740a 100644
--- a/drivers/usb/chipidea/udc.c
+++ b/drivers/usb/chipidea/udc.c
@@ -2051,7 +2051,8 @@ static void udc_id_switch_for_host(struct ci_hdrc *ci)
 	 * (in otg fsm mode, means B_IDLE->A_IDLE) due to ID change.
 	 */
 	if (!ci_otg_is_fsm_mode(ci) ||
-			ci->fsm.otg->state == OTG_STATE_B_IDLE)
+			ci->fsm.otg->state == OTG_STATE_B_IDLE ||
+			ci->fsm.otg->state == OTG_STATE_UNDEFINED)
 		hw_write_otgsc(ci, OTGSC_BSVIE | OTGSC_BSVIS, OTGSC_BSVIS);
 
 	ci->vbus_active = 0;
-- 
1.7.9.5

